specification/transport_layer/transport_layer.tex

```diff
@@ -575,6 +575,72 @@ \subsection{Transfer-ID comparison}\label{sec:transfer_id_forward_distance}
 \end{minted}
 \end{minipage}
 
+\subsection{Payload truncation}
+
+The deterministic nature of UAVCAN in general and DSDL in particular allows implementations to statically determine the
+maximum amount of memory that is required to contain a data object of a particular type.
+Consequently, an implementation that is interested in receiving data objects of a particular type\footnote{%
+    Messages, service requests, or service responses.
+}
+can statically determine the maximum length of the transfer payload.
+
+Implementations shall be able to handle incoming transfers containing a larger amount of payload data than expected.
+In the event of such extra payload being received, a compliant implementation should silently\footnote{%
+    Such occurrence is not indicative of a problem so it shall not be reported as such.
+}
+discard the excessive (unexpected) data at the end of the received payload.
+The transfer CRC, if applicable, shall be validated regardless of the presence of the extra payload in the transfer.
+
+The requirement to silently discard the excessive payload data at the end of the transfer is motivated by
+the necessity to allow extensibility of data type definitions, as described in chapter \ref{sec:dsdl}.
+Additionally, excessive payload data may contain padding bytes if required by the particular transport layer.
+
+\begin{figure}[H]
+    $$
+    \raisebox{1em}{\footnotesize{\text{first byte}}}
+    \overbrace{%
+        \underbrace{\huge{%
+            \blacksquare\blacksquare\blacksquare\blacksquare\blacksquare\blacksquare%
+            \blacksquare\blacksquare\blacksquare\blacksquare\blacksquare\blacksquare%
+        }}_{\substack{\text{Expected, accepted} \\ \text{payload}}}%
+        \underbrace{\huge{%
+            \boxtimes\boxtimes\boxtimes\boxtimes\boxtimes\boxtimes\boxtimes\boxtimes%
+        }}_{\substack{\text{Excessive, discarded} \\ \text{payload}}}%
+    }^{\substack{%
+        \text{Transfer CRC is validated} \\
+        \text{for the entire transfer payload} \\
+        \text{before the truncation}}
+    }
+    \raisebox{1em}{\footnotesize{\text{last byte}}}
+    $$
+    \caption{Transfer payload truncation.\label{fig:transport_payload_truncation}}
+\end{figure}
+
+\begin{remark}
+    Let node $A$ publish an object of the following type over the subject $X$:
+
+    \begin{minted}{python}
+        float32 parameter
+        float32 variance
+    \end{minted}
+
+    Let node $B$ subscribe to the subject $X$ expecting an object of the following type:
+
+    \begin{minted}{python}
+        float32 parameter
+    \end{minted}
+
+    The payload truncation requirement guarantees that the two nodes will be able to interoperate despite
+    relying on incompatible data type definitions.
+    Under this example, the duty of ensuring the semantic compatibility lies on the system integrator.
+\end{remark}
+
+Implementations shall not enforce the minimum payload size on received transfers.
```

Comment:
An example here for handling this case would be illuminating.

Reply:
I just made a remark out of this with a reference to the DSDL serialization section.